packages-exp/auth-exp/src/core/user/token_manager.test.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import { StsTokenManager, DATE_GENERATOR } from './token_manager';
+import { IdTokenResponse } from '../../model/id_token';
+import { createSandbox } from 'sinon';
+
+use(chaiAsPromised);
+
+const sandbox = createSandbox();
+
+describe('core/user/token_manager', () => {
+  let stsTokenManager: StsTokenManager;
+  let now: number;
+
+  beforeEach(() => {
+    stsTokenManager = new StsTokenManager();
+    now = Date.now();
+    sandbox.stub(DATE_GENERATOR, 'now').returns(now);
+  });
+
+  afterEach(() => sandbox.restore());
+
+  describe('#isExpired', () => {
+    it('is true if past expiration time', () => {
+      stsTokenManager.expirationTime = 1; // Ancient history
+      expect(stsTokenManager.isExpired).to.eq(true);
+    });
+
+    it('is true if exp is in future but within buffer', () => {
+      // Buffer is 30_000
+      stsTokenManager.expirationTime = now + 20_000;
```

Comment:
is it possible to pull in the buffer as a variable here?

Reply:
Done